### DISCUSSION TITLE
Is this the new StackOverflow?

### DISCUSSION THREAD

**Discussion by axelrindle** [community/community#3691] [Discussions] [answered]
Github Discussions looks like a clone of StackOverflow. Question voting, accepted answers, all of that stuff. I'm not sure if Github is the right platform to introduce such a feature, especially when there seems to be no true delimitation between Issues and Discussions.

**Comment by DipLeChip** [comment]
its good for repos tho, say like idk any repo if they have one, people can start a discussion on issues, suggestions etc.

**Comment by PranamBhat** [comment]
@axelrindle  To be honest, it not! **GitHub** has it's own way of serving the community with great aim.

**Reply by MrCrypticxDev** [reply]
GitHub is more interactive, you can deal with stuff and post your code!

**Comment by swtanggara** [comment]
I think GitHub Discussions is a feature to defeat what is offered by [Stackoverflow for Teams](https://stackoverflow.com/teams), which I recently came up there from this [blog post](https://stackoverflow.blog/2021/03/17/stack-overflow-for-teams-is-now-free-forever-for-up-to-50-users/).

The lack feature of this GitHub Discussions I think is **labeling/tagging** to give the users more granular grouping or search capabilities.

**Reply by evi-liu** [reply]
Hi @swtanggara, when you say "give the users more granular grouping or search capabilities", what kind of users are you referring to (e.g. maintainers, authors, etc)?

**Reply by swtanggara** [reply]
I think all kinds of users. The context is to be able to tagging/labelling so we can filters the topic of discussions. I wish this could close the gap with Stackoverflow.

**Reply by TonyGravagno** [reply]
It's not necessary to compare with Stack Overflow as a metric. Every environment benefits from tagging and cross-linking. For example, if I start a discussion about how to use Regex with some library, someone can tag the discussion as being about "filters" or "search", or "regular expressions". I might just ask how to filter data and the response might be a terse "use this code", with no mention of Regex. A repo owner or non-owner developer should be able to quickly find discussions on this topic of how people are using the code, without being restricted by the vocabulary or linguistic skills that were used in the content. That's part of what metadata is for, supplementing the content.

In the bigger picture, we've used tags for decades because we lacked AI technology that can derive context from content. If searching is supplemented by AI, so that a search for "filters" reveals a discussion on Regex, without tags or specific keywords, then we won't need to rely on tags so much.

**Comment by TonyGravagno** [comment, marked as answer]
Stack Overflow is focused on **Questions and Answers**. A **Discussion** is an exchange of comments, opinions, ideas, and recommendations, which are strongly discouraged in SO. SO is **not** a discussion environment. You'll get downvoted in SO for asking the "best" way use code or for alternatives to some tagged solution. We do not, or should not, give upvotes = points in SO for a well-stated comment, only for solid fact-based answers to specific points that contribute to the OP question.

This Discussion environment is not the same as SO and it is not the same as Issues. We have used Issues for Q&A and discussions because there was no other place for it. Each of these environments has unique value. I don't like the idea that people think these are competing environments. I don't want SO to open up to random discussions because it will bury solutions in chit-chat. And I don't want to get downvoted in a discussion for an unpopular choice of code style or an opinion about deprecating a feature. I also don't want some repo owner to Close what I believe is a productive discussion about an ongoing challenge with their software - closing an issue ends the discussion, which is **not** the way of "free and open source".

The issue we see in this thread and others is that people confuse these topics, Q&A, Issues, and Discussions. We've been trying for so many years to bang each of these nails with the same hammer, that now people don't know what a discussion is anymore. They think discussion=issue, or Q&A=discussion, or issue=Q&A.  And yet we clearly get reprimanded and/or find dis-satisfaction in any of these environments by random administrators when we don't use the environment as they believe the environment is rigidly defined.

If anything, Github Discussions are more like Google Groups, Forums (phpBB, Xenforo, etc), G+, or even Facebook pages and groups. The advantage of Discussions in this environment is that we can be more focused with meaningful links to local assets and whatever other features they add to bond these tools.

So, um, "no", IMO this is **not** the new StackOverflow, and the fact that a question like this is asked is surely a sign that we need a tool like this to help our culture re-learn what it means to have a discussion on some topic. (That's no slam to anyone, sincerely, I'm just pointing out that there is something deeper to the question.)

**Comment by swtanggara** [comment]
Yay, it's now has tagging/labelling 😁

**Reply by evi-liu** [reply]
yes, try it out and let us know what you think!

**Reply by ghalimi** [reply]
I think GitHub Discussions are awesome. What's missing though are true tags, not just labels. Don't get me wrong, labels are great, but they cannot be used as tags. Labels are part of the schema, and are best when 20 or less are used, since they use colors (you can't really find a well-contrasted color palette with more than 20 different colors). For their part, tags should be part of the data, and define a taxonomy. There could be hundreds or even thousands of tags.
